chore(CDC): [IOBP-1010] Remove CDC Bonus files and related functions

LeleDallas opened the pull request:
## Short description
This pull request involves the removal of several components and related functionalities for the CDC bonus feature. The most important changes include the deletion of tracking logic, API client code, UI components, and associated tests.

## List of changes proposed in this pull request
- Removed CDC Bonus files inside `cdc` folder
- Removed CDC Bonus saga, store, navigation and analytics

## How to test
Ensure the application is not crashing

Hantex9 reviewed (commented):
That's awesome! 🧹

What do you think also removing every definitions reference even inside the `generate-api-models.sh` file and removing the `assets/CdcSwagger.yml` file?

LeleDallas commented:
> That's awesome! 🧹
> 
> What do you think also removing every definitions reference even inside the `generate-api-models.sh` file and removing the `assets/CdcSwagger.yml` file?

Addressed here: cdbd94802a2686fe811be7bdc692205fef78c598
In this commit b6be0d27116eceb4d9f582ddbbb46f5e93fda748 I also removed the related locations

dmnplb commented:
What do you think about removing the `RadioButtonList` component? It's a legacy component that is only used in the CDC flow.

LeleDallas commented:
> What do you think about removing the `RadioButtonList` component? It's a legacy component that is only used in the CDC flow.

🧹 here 8d131bb721f8d219d65f83b58623486e10d9f3f8

Hantex9 reviewed (approved):
LGTM! 🚀
